lib/Controller/SlaveController.php

```diff
@@ -178,23 +178,40 @@ public function autoLogin($jwt) {
 	/**
 	 * Create app token
 	 *
+	 * @PublicPage
+	 * @NoCSRFRequired
 	 * @NoAdminRequired
 	 *
 	 * @return DataResponse
 	 */
-	public function createAppToken() {
+	public function createAppToken($jwt) {
 
-		if($this->gss->getMode() === 'master') {
+		if($this->gss->getMode() === 'master' || empty($jwt)) {
 			return new DataResponse([], Http::STATUS_BAD_REQUEST);
 		}
 
-		$user = $this->userSession->getUser();
-		$uid = $user->getUID();
-
-		$token = $this->tokenHandler->generateAppToken($uid);
+		try {
+			list($uid, $password, $options) = $this->decodeJwt($jwt);
 
-		return new DataResponse($token);
+			if ($this->userManager->userExists($uid)) {
+				// if we have a password, we verify it
+				if (!empty($password)) {
+					$result = $this->userSession->login($uid, $password);
+				} else {
+					$result = true;
+				}
+				if ($result) {
+					$token = $this->tokenHandler->generateAppToken($uid);
```

Comment:
Is there a reason not the include the password if it is send?

Reply:
I also wondered if I could include the password. At the end I decided to stick to the original code, which also didn't used the password, in order to avoid any problems.

Reply:
The only thing missing would be that external storage can't use the users password etc

However. I'm totally fine with that being a GS limitation.
It is anyways a limitation of SSO (as we never get the password). And also in a GS settings it is questionable if this is wanted.